src/nvim/os/env.c

```diff
@@ -187,45 +187,63 @@ void init_homedir(void)
 
   const char *var = os_getenv("HOME");
 
-#ifdef WIN32
-  // Typically, $HOME is not defined on Windows, unless the user has
-  // specifically defined it for Vim's sake. However, on Windows NT
-  // platforms, $HOMEDRIVE and $HOMEPATH are automatically defined for
-  // each user. Try constructing $HOME from these.
-  if (var == NULL) {
-    const char *homedrive = os_getenv("HOMEDRIVE");
-    const char *homepath = os_getenv("HOMEPATH");
-    if (homepath == NULL) {
-      homepath = "\\";
-    }
-    if (homedrive != NULL
-        && strlen(homedrive) + strlen(homepath) < MAXPATHL) {
-      snprintf(os_buf, MAXPATHL, "%s%s", homedrive, homepath);
-      if (os_buf[0] != NUL) {
-        var = os_buf;
+#ifdef UNIX
+  // use uv_os_homedir only for unix systems
+  // as uv_os_homedir checks %USERPROFILE% first then only %HOMEDRIVE%%HOMEPATH%  
+  // uv_os_homedir doesn't work when a buffer initialized to NULL is passed 
+  // so using a local buffer
```

Comment:
"buffer initialized to NULL" is a long-winded way of saying "NULL" ?

Reply:
Yes. Just wanted to point out that  ```static char* homedir = NULL ``` would cause `uv_os_homedir()` to fail if `homedir` was passed directly to the function.